packages/core/src/JWEditor.ts

```diff
@@ -333,6 +334,24 @@ export class JWEditor {
         // todo: Improve mode to use ReactiveValue when it will be implemented.
         this.plugins.get(Mode).setCurrent(modeId);
     }
+
+    /**
+     * Return true if the given node is editable.
+     *
+     * @param node
+     */
+    isEditable(node: VNode): boolean {
+        return this.mode ? this.plugins.get(Mode).isEditable(node) : node.editable;
+    }
+
+    /**
+     * Return true if the given node is breakable.
+     *
+     * @param node
+     */
+    isBreakable(node: VNode): boolean {
+        return this.mode ? this.plugins.get(Mode).isBreakable(node) : node.breakable;
+    }
```

Comment:
Verry strange to have that kind of function in the editor itself. It ties the mode to the editor even more.
It's another argument in favor to pass only the mode to the objects that need mode functionnality.

We should avoid to make "proxy" method like that on `JWEditor`.

Also, I believe that `this.mode` should never be undefined. There should always have a mode that is active. Otherwise it will require us to make these kind of branching.

Keep in mind that in the future, the mode could set any kind of properites for the node.

Reply:
Well I'd argue that it's the _editor_'s mode so to me it makes sense.

I removed your default mode because that was the reason you added a `checkEditable` key in the mode, so we could use the `editable` key of the node. That means we'd have to do that for every property that the mode could set...
But come to think of it, that `checkEditable` could maybe be replaced with a rule, like `{ selectors: [ node => node.editable ], editable: true }`.

So in conclusion I guess I could pass the `Mode` plugin wherever it's needed and have a default mode that just has the line above. Then I could remove these two functions.

@dmo-odoo Do you also think that would be better?

Reply:
I forgot to tell in the previous comments, the mode class should not be a plugin, it should be the class provided by the `ModePlugin` (I don't know how to name thoses because of the current convention of plugin naming).

As discussed with @dmo-odoo, you could have multiples range in the dom that are not in the same mode. (e.g. collaboration).

I don't understand your argument about adding a `checkEditable` for all keys as you do it with `isEditable`, `isBreakable`.

Reply:
I changed these things from your first draft for a reason. I renamed `ModePlugin` into `Mode` because of said conventions of plugin naming. Then what you named `Mode` was replaced with `ModeDescription` which is enough to describe a mode. A class is overkill imo as it's purely declarative.

The problem with `checkEditable` is that it's a boolean that informs the plugin of whether it should ignore the mode for a given property of not. `isEditable`, on the other hand, implements a real feature. And as you mentioned above the logic is duplicated between `isEditable` and `isBreakable` so technically I could perfectly do something like `is(node, 'editable')`, `is(node, 'breakable')`, `is(node, 'tangible')` etc. instead.

In any case I countered my own argument just below it when I said I _could_ have a default mode with a `{ selectors: [ node => node.editable ], editable: true }` rule to replace `checkEditable: false` so... :shrug:

Reply:
It's funny because it *seems* we yet again have three different views on this. @Zinston obviously likes the current version since he considers he's done with this PR. @Goaman wants the `Mode` to stay in a plugin and pass the mode rather than the editor (I guess so because you mentionned a `ModePlugin`). And I think I would put the `Mode` in the core of the editor and stop pretending it's a plugin since it is used everywhere and most notably in `VRange`.